h/static/styles/partials-v2/_form-input.scss

```diff
@@ -50,6 +50,38 @@
     color: $grey-5;
   }
 
+  .form-input__character-counter {
+    @include font-normal;
+
+    // Show fallback "counter" if loading js times out.
+    .env-js-timeout & {
+      display: initial;
+    }
+
+    // Immediately hide counter if js capable - prevents momentary flash of
+    // unenhanced fallback counter before js kicks in.
+    .env-js-capable & {
+      display: none;
+    }
+
+    // The js controller adds this class when it's ready - show the now
+    // enhanced counter.
+    &.is-ready {
+      display: initial;
+    }
+
+    position: absolute;
+    bottom: 10px;
+    right: $h-padding;
+
+    color: $grey-5;
+  }
+
+  .form-input__character-counter--error {
```

Comment:
`--foo` classes are for _static_ modifications to a class, eg. `btn--cancel` or `btn--danger`. For dynamic _states_ applied via JS, use `is-$state` classes with combination selectors, like `.is-ready` above. See https://github.com/hypothesis/frontend-toolkit/blob/master/docs/css-style-guide.md#naming